dnscache.go

```diff
@@ -6,22 +6,27 @@ import (
 	"sync"
 	"time"
 
-	"github.com/rs/dnscache/internal/singleflight"
+	"golang.org/x/sync/singleflight"
 )
 
+type DNSResolver interface {
```

Comment:
Let's call it `Resolver`, DNS is redondant with the package name.

Reply:
Well, that makes sense.  I change the name of the interface immediately to `Resolver`

Reply:
You're right, what about NetResolver?

Reply:
I guess DNSResolver is better, let's keep it that way.